document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoSortingExpressionParser.java

```diff
@@ -9,51 +15,67 @@
 import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
 import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
 import org.hypertrace.core.documentstore.expression.operators.SortingOrder;
-import org.hypertrace.core.documentstore.parser.SortingExpressionParser;
-import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.parser.SortingExpressionVisitor;
+import org.hypertrace.core.documentstore.query.QueryInternal;
 import org.hypertrace.core.documentstore.query.SortingSpec;
 
-public class MongoSortingExpressionParser extends MongoExpressionParser
-    implements SortingExpressionParser {
+public final class MongoSortingExpressionParser implements SortingExpressionVisitor {
 
   private static final String SORT_CLAUSE = "$sort";
+  private static final Map<SortingOrder, Integer> ORDER_MAP =
+      unmodifiableMap(
+          new EnumMap<>(SortingOrder.class) {
+            {
+              put(ASC, 1);
+              put(DESC, -1);
+            }
+          });
 
   private final SortingOrder order;
 
-  protected MongoSortingExpressionParser(final Query query, final SortingOrder order) {
-    super(query);
+  MongoSortingExpressionParser(final SortingOrder order) {
     this.order = order;
   }
 
+  @SuppressWarnings("unchecked")
   @Override
-  public Map<String, Object> parse(final AggregateExpression expression) {
+  public Map<String, Object> visit(final AggregateExpression expression) {
     throw new UnsupportedOperationException(
         String.format(
             "Cannot sort an aggregation ($%s) in MongoDB. "
                 + "Set alias in selection and sort by the alias as identifier",
             expression.getAggregator().name().toLowerCase()));
   }
 
+  @SuppressWarnings("unchecked")
   @Override
-  public Map<String, Object> parse(final FunctionExpression expression) {
+  public Map<String, Object> visit(final FunctionExpression expression) {
     throw new UnsupportedOperationException(
         String.format(
-            "Cannot sort a function ($%s) in MongoDB.",
+            "Cannot sort a function ($%s) in MongoDB."
+                + "Set alias in selection and sort by the alias as identifier",
             expression.getOperator().name().toLowerCase()));
   }
 
+  @SuppressWarnings("unchecked")
   @Override
-  public Map<String, Object> parse(final IdentifierExpression expression) {
-    String parsed = new MongoIdentifierExpressionParser(query).parse(expression);
-    return Map.of(parsed, getOrder());
+  public Map<String, Object> visit(final IdentifierExpression expression) {
+    String parsed = new MongoIdentifierExpressionParser().parse(expression);
+    Integer value = ORDER_MAP.get(order);
+
+    if (value == null) {
```

Comment:
I think we can check this beginning of the method before parsing. If the operator is not supported, we fail fast.

Reply:
Done.